### PR TITLE
SI-7185 Avoid NPE in TreeInfo.isExprSafeToInline

### PR DESCRIPTION
**Pull request by retronym**
We got there typechecking code with a redundant
layer of Block.

We can't express that in source code, so we test
this with manual tree construction and with XML
literals, which as reported produce such trees.

Review by @xeno-by

### CODE REVIEW

File: src/reflect/scala/reflect/internal/TreeInfo.scala [hunk 1]
```diff
@@ -98,7 +98,8 @@ abstract class TreeInfo {
       // However, before typing, applications of nullary functional values are also
       // Apply(function, Nil) trees. To prevent them from being treated as pure,
       // we check that the callee is a method.
-      fn.symbol.isMethod && !fn.symbol.isLazy && isExprSafeToInline(fn)
+      // The callee might also be a Block, which has a null symbol, so we guard against that (SI-7185)
+      fn.symbol != null && fn.symbol.isMethod && !fn.symbol.isLazy && isExprSafeToInline(fn)
```
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/TreeInfo.scala:102]: Isn't the `fn.symbol` check a bit too restrictive?

Comment (retronym) [inline src/reflect/scala/reflect/internal/TreeInfo.scala:102]: The part I added, or the existing code?

Comment (xeno-by) [inline src/reflect/scala/reflect/internal/TreeInfo.scala:102]: The part you added. What about `{ safeToInline }`?

Comment (retronym) [inline src/reflect/scala/reflect/internal/TreeInfo.scala:102]: Can you tell me what `Apply(???, _)` tree you like to have deemed safe to inline, where `???` is a symbol-less tree node?

Comment (xeno-by) [inline src/reflect/scala/reflect/internal/TreeInfo.scala:102]: What if it's a Block, whose stats are empty and whose expr is a safe to inline tree?

Comment (retronym) [inline src/reflect/scala/reflect/internal/TreeInfo.scala:102]: false negatives don't really hurt us much here.

**Comment by paulp** [comment]
LGTM.

**Pull request merged**